wagtail/wagtailadmin/templates/wagtailadmin/pages/edit.html

```diff
@@ -99,4 +99,17 @@ <h1 class="icon icon-doc-empty-inverse">{% blocktrans with title=page.title page
 {% block extra_js %}
     {{ block.super }}
     {% include "wagtailadmin/pages/_editor_js.html" %}
+
+    <script>
+        $(function() {
+            /* Make user confirm before leaving the editor if there are unsaved changes */
+            enableDirtyFormCheck(
+                function() {
+                    return $('#page-edit-form').serialize()
+                },
+                '{% trans "This page has unsaved changes." %}',
```

Comment:
Are you sure that `trans` will return correct JS string (without `'` character, for example) on all languages? Maybe we should apply `escapejs` filter here?

If you will decide to apply `escapejs`, please check how browser prompt will work with non-ANSI characters.